File: app/src/main/java/org/techtown/find_gas_station/Util/Comparator/OilDistanceComparator.java
package org.techtown.find_gas_station.Util.Comparator;

import org.techtown.find_gas_station.OilList;

import java.util.Comparator;

public class OilDistanceComparator implements Comparator<OilList> {
    @Override
    public int compare(OilList t1, OilList t2) {

        if(Integer.parseInt(t1.distance) == Integer.parseInt(t2.distance)){
            return Integer.parseInt(t1.price)-Integer.parseInt(t2.price);
        }
        else{
            return Integer.parseInt(t1.distance) - Integer.parseInt(t2.distance);
        }
    }
}
//거리순 오름차순 정렬
//만약 거리가 같다면 가격이 같은 순으로 만들기

File: app/src/main/java/org/techtown/find_gas_station/Util/Comparator/OilDistanceComparator.kt
package org.techtown.find_gas_station.Util.Comparator

import org.techtown.find_gas_station.Data.TotalOilInfo

class OilDistanceComparator : Comparator<TotalOilInfo> {
    override fun compare(t1 : TotalOilInfo, t2 : TotalOilInfo) =
        if (t1.distance.toInt() == t2.distance.toInt()) {
            t1.price.toInt() - t2.price.toInt()
        } else {
            t1.distance.toInt() - t2.distance.toInt()
        }


}